s3torchconnector/src/s3torchconnector/dcp/s3_file_system.py

```diff
@@ -22,17 +22,20 @@
     FileSystemWriter,
     FileSystemBase,
 )
+import torch
 
 from s3torchconnector._s3client import S3Client
 from s3torchconnector._s3dataset_common import parse_s3_uri
+from .._user_agent import UserAgent
 
 logger = logging.getLogger(__name__)
 
 
 class S3FileSystem(FileSystemBase):
     def __init__(self, region: str, s3_client: Optional[S3Client] = None) -> None:
         self._path: Union[str, os.PathLike] = ""
-        self._client = s3_client if s3_client is not None else S3Client(region)
+        self.user_agent = UserAgent(["dcp", torch.__version__])
```

Comment:
Do we need to have a `self.user_agent` field? Or could we just pass it to the `S3Client` below? E.g.:

```python
self._client = s3_client if s3_client else S3Client(region=region, user_agent=UserAgent(["dcp", torch.__version__]))
```

Also, shall we unit test this?

Reply:
Made `user_agent` as variable and added tests